apps/gp2-server/src/data-providers/working-group.data-provider.ts

```diff
@@ -32,15 +56,53 @@ export class WorkingGroupSquidexDataProvider
     return {
       total: result.queryWorkingGroupsContentsWithTotal.total,
       items: result.queryWorkingGroupsContentsWithTotal.items.map(
-        (workingGroup) => ({
-          ...workingGroup.flatData,
-          id: workingGroup.id,
-          title: workingGroup.flatData.title || '',
-          shortDescription: workingGroup.flatData.shortDescription || '',
-          leadingMembers: workingGroup.flatData.leadingMembers || '',
-          members: [],
-        }),
+        parseWorkingGroupToDataObject,
       ),
     };
   }
 }
+type Member = NonNullable<GraphQLWorkingGroup['flatData']['members']>[number];
+
+const parseWorkingGroupMembers = (
```

Comment:
this function's signature feels a little odd. It's been written this way only to work with reduce. It would make so much more sense if it accepted an input, which should be a `Member` (it should actually be something else but I'll get to that) and and returned a single item of type `gp2.WorkingGroupMember`.

In order to do that I would keep the reduce elsewhere and have this mapper/parser used inside the reduce.

Reply:
done